invenio_app_ils/ui/src/invenio_app_ils/forms/core/VocabularyField.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { vocabulary as vocabularyApi } from '../../common/api';
+import { SelectField } from './SelectField';
+import { invenioConfig } from '../../common/config';
+import { AccordionField } from './AccordionField';
+
+export class VocabularyField extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoading: true,
+      error: null,
+      options: [],
+    };
+
+    this._mounted = false;
+  }
+
+  query = () => {
+    const searchQuery = vocabularyApi
+      .query()
+      .withType(this.props.type)
+      .withSize(invenioConfig.max_results_window)
+      .qs();
+    return vocabularyApi.list(searchQuery);
+  };
+
+  serializer = hit => ({
+    key: hit.metadata.id,
+    value: hit.metadata.key,
+    text: hit.metadata.text,
+  });
+
+  fetchVocabularies = async serializer => {
+    try {
+      const response = await this.query();
+      const options = [];
+      if (response.data.hits.length > 0) {
+        options.push({
+          key: '',
+          value: '',
+          text: '-',
+        });
+      }
+      for (let hit of response.data.hits) {
+        options.push(serializer(hit));
+      }
+      if (this._mounted) {
+        this.setState({ isLoading: false, options: options, error: null });
+      }
+    } catch (error) {
+      this.setState({
+        isloading: false,
+        options: [{ key: '', value: '', text: 'Failed to load vocabularies.' }],
+        error: {
+          content: 'Failed to load vocabularies.',
+          pointing: 'above',
+        },
+      });
+    }
+  };
+
+  componentDidMount() {
+    this._mounted = true;
+    const serializer = this.props.serializer || this.serializer;
+    this.fetchVocabularies(serializer);
+  }
+
+  componentWillUnmount() {
+    this._mounted = false;
+  }
+
+  render() {
+    const {
+      accordion,
+      fieldPath,
+      label,
+      multiple,
+      serializer,
+      type,
+      ...uiProps
+    } = this.props;
+    const { isLoading, options } = this.state;
+    const noResultsMessage = isLoading
+      ? 'Loading vocabularies...'
```

Comment:
for the user's perspective I would display `Loading options...` I don't think they will know what is a vocabulary